ansible/roles/kubernetes/tasks/debian.yml

```diff
@@ -0,0 +1,39 @@
+---
+# Instructions: https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#installing-kubeadm-kubelet-and-kubectl
+- name: add apt signing key for kubernetes (1/4)
+  apt_key:
+    url: https://packages.cloud.google.com/apt/doc/apt-key.gpg
+    state: present
+
+# Removed the following repo due to stability issues: https://apt.kubernetes.io/
+- name: adding apt repository for kubernetes (2/4)
+  apt_repository:
+    repo: deb http://packages.cloud.google.com/apt/ kubernetes-xenial main
+    state: present
+  register: kubernetes_repository
+  retries: 10
+  until: kubernetes_repository is success
+
+# To downgrade these packages use the following apt options:
+#   dpkg_options: force-downgrade
+#   force: yes
+- name: install kubernetes packages (3/4)
+  apt:
+    name:
+      - kubelet={{ kubelet_version }}
+      - kubeadm={{ kubeadm_version }}
+      - kubectl={{ kubectl_version }}
+    state: present
+    update_cache: yes
+  register: apt_install_kube
+  retries: 5
+  until: apt_install_kube is success
+
+- name: add apt-mark hold to avoid version skew issues. (4/4)
+  dpkg_selections:
+    name: "{{ item }}"
+    selection: hold
+  with_items:
```

Comment:
`with_items` can be replaced by `loop`. Both work fine and `with_items` is not yet deprecated, but might be in the furure. See the "Note" at: https://docs.ansible.com/ansible/latest/user_guide/playbooks_loops.html